### DISCUSSION TITLE
Would like to use Grapevine v5 with my .Net 4.6 project

### DISCUSSION THREAD

**Discussion by Lroy** [scottoffen/grapevine#40] [Q&A] [answered]
I have a rather large project that is currently using .Net Framework 4.6.1

I am successfully using Grapevine 4.2.2 to prototype some concepts.

Is Grapevine actually dependent on .Net 5.0 (would be involved to port it down to 4.6.1)?

**Comment by scottoffen** [comment, marked as answer]
Grapevine 5 currently targets the following frameworks:

- .Net Standard 2.0
- .Net Standard 2.1
- .Net 5.0

While I've not explicitly tried it, according to the [.Net Standard documentation](https://docs.microsoft.com/en-us/dotnet/standard/net-standard), .Net 4.6.1 _should_ be able to use Grapevine 5. Have you tried to use it yet?

**Edit:** I just spun up a small console application targeting .Net Framework 4.6.1, pulled in the latest release candidate for Grapevine 5, and started a server and hit my basic route. Unless you are experiencing a specific problem, I'd say that **you can use Grapevine 5 with .Net Framework 4.6.1**.

**Comment by Lroy** [comment]
I have not tried it, because I also have a VS 2015 project format issue, where I would need to somehow re-format the SLN and CSPROJ files to conform to your VS tools version.

I have been using V4, and I must say that it works very nicely for what I need to do!

Having a slight issue getting SSL/HTTPS to work, but I’m fairly certain that the problem has to do with the myriad steps required to get certs in the right places…

Thank you for creating such a well-thought-out REST library!

**Reply by scottoffen** [reply]
I'm glad you find it useful! Nevertheless, I'm sad that you're using Grapevine 4. Grapevine 5 is so much better. 😄

**Reply by Lroy** [reply]
Does anyone have a precise series of steps required to access a Grapevine-based listener at a specific IP address and port?

I have tried basically everything that people suggest doing.  I still don't have access to my HTTPS server.

I'm not actually positive that modern browsers will access an HTTPS server at a specific IP address.  Most general articles talk about the requirement for a domain name to gain access to a server, while fixed-ip addresses are not allowed apparently.

Anyway, this is what I currently have done:

- netsh http add urlacl url="https://+:80/" user="everyone" (this command worked)
- powershell -Command "New-SelfSignedCertificate -DnsName localhost -CertStoreLocation cert:\LocalMachine\My -NotAfter (Get-Date).AddYears(10)" (this command worked)
- Copied cert to "Trusted Root Certification Authorities->Certificates" in mmc (the cert looks good)
- netsh http add sslcert ipport=localhost:80 certhash=<thumbprint> appid={my app GUID} (this command worked)
- My Grapevine example server is listening at "0.0.0.0", port 80. (server.UseHttps = true)
-    Server ListenerPrefix = "https://+:80/"

ACL, Cert generation, and SSL binding all seem to be reasonable.

When I try to access the server on the same machine (using IP or 'localhost') as the Grapevine server, I get "ERR_CONNECTION_REFUSED" in the browser (Chrome).

If I try to access the server from outside the server machine, I get identical results.

Is HTTPS even possible without using a registered domain name and official CA?

Thanks for any help!

Larry

**Reply by Lroy** [reply]
Update:

Got my Grapevine server to work!

Changes:
- netsh http add urlacl url="https://+:443/" user="everyone" (this command worked)
- powershell -Command "New-SelfSignedCertificate -DnsName localhost -CertStoreLocation cert:\LocalMachine\My -NotAfter (Get-Date).AddYears(10)" (this command worked)
- Copied cert to "Trusted Root Certification Authorities->Certificates" in mmc (the cert looks good)
- netsh http add sslcert ipport=0.0.0.0:443 certhash= appid={my app GUID} (this command worked)
- My Grapevine example server is listening at "0.0.0.0", port 443. (server.UseHttps = true)
- Server ListenerPrefix = "https://+:443/"

Maybe I cannot have an HTTPS server at port 80 using the common methodology?

**Reply by scottoffen** [reply]
In your first post, you used this command:
```
netsh http add sslcert ipport=localhost:80 certhash= appid={my app GUID}
```

And in your next post, you used the command:
```
netsh http add sslcert ipport=0.0.0.0:443 certhash= appid={my app GUID}
```

While the port did change, I would think it more likely that the change from `localhost` to `0.0.0.0` was what made the difference.

Glad you got it working!

**Reply by scottoffen** [reply]
By the way, there is a similar discussion board for GV4 available. If that is what you are using, you might want to post your questions over there. Granted, your question here is independent of the version of Grapevine you're using.

https://github.com/scottoffen/grapevine-legacy/discussions

**Comment by Lroy** [comment]
Got it.  That’s a good idea.

I now have everything working, except that I’m working my way through how to get rid of Chrome messages regarding accessing the Grapevine site on a local network, by IP address.  The DnsName given in the CERT is not going to match the address in the Chrome URL, so Chrome (and others) complain about the site not being secure.

Not sure how to solve this one, actually, because I deliver custom industrial systems, and they will never have certs from a CA…

Larry


> On Jan 30, 2021, at 5:11 PM, Scott Offen <notifications@github.com> wrote:
> 
> 
> By the way, there is a similar discussion board for GV4 available. If that is what you are using, you might want to post your questions over there.
> 
> https://github.com/scottoffen/grapevine-legacy/discussions <https://github.com/scottoffen/grapevine-legacy/discussions>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/scottoffen/grapevine/discussions/40#discussioncomment-324131>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AACG3AR6B63M5HNDIR237FLS4R7Y3ANCNFSM4WX5TLNA>.
> 

Thank You,

Larry Roy
Embedded Logix, Inc.
288 S Main
Romeo MI 48065
USA







CONFIDENTIALITY NOTICE: This communication and/or transmission (including any attachments) is intended solely for the person or entity to which it is addressed and may contain confidential, trade secret, privileged, and/or proprietary information and materials. Any review, distribution, dissemination, printing, copying, exploitation, or other use or communication of this email and/or its contents by persons or entities other than the specific intended recipient or his/her duly authorized representative is strictly prohibited. If you have received this communication in error, please contact the sender immediately and delete the communication from your computer and all other devices and related systems and shred and properly discard any tangible versions. Thank you.
